runtime.md

```diff
@@ -34,22 +32,25 @@ See [Query State](#query-state) for information on retrieving the state of a con
-4. The user specified process MUST be executed in the container.
-5. The poststart hooks MUST be invoked by the runtime.
-   If any poststart hook fails, then the container MUST be stopped and the lifecycle continues at step 7.
-6. Additional actions such as pausing the container, resuming the container or signaling the container MAY be performed using the runtime interface.
-   The container MAY also error out, exit or crash.
-7. The container MUST be destroyed by undoing the steps performed during create phase (step 2).
-8. The poststop hooks MUST be invoked by the runtime and errors, if any, SHOULD be logged.
-
-Note: The lifecycle is a WIP and it will evolve as we have more use cases and more information on the viability of a separate create phase.
+1. OCI compliant runtime's `create` command is invoked with a reference to the location of the bundle and a unique identifier.
+   How these references are passed to the runtime is an implementation detail.
+2. The container's runtime environment (namespaces, mounts, etc.) MUST be created according to the configuration in [`config.json`](config.md).
+   If a new PID namespace is requested in the [`config.json`](config.md), a PID namespace MUST created at this time.
+   However, the user-specified code (from [`process`](config.md#process-configuration) MUST NOT be run at this time.
+   With the exception of the user-specified process, any updates to `config.json` after container is created MUST NOT affect the container.
+3. Once the container is created additional actions MAY be performed based on the features the runtime chooses to support.
+  However, some actions might only be available based on the current state of the container (e.g. only available while it is started).
+4. Runtime's `start` command is invoked with the unique identifier of the container.
+   The runtime MUST run the user-specified code, as specified by [`process`](config.md#process-configuration), in the container's PID namespace.
+   Any updates to the user-specified code in [`process`](config.md#process-configuration) after this point MUST NOT have any effect on the container.
+5. The container's processes are stopped.
+   This MAY happen due to them erroring out, exiting or crashing.
+   The runtime's `stop` operation MAY also be used to stop the processes running within the container.
+6. If the [`autoremove`](config.md#root-configuration) property is present with a value of `true` the runtime's `delete` operation is automatically invoked.
+6. Runtime's `delete` command is invoked with the identifier of the container.
+   The container MUST be destroyed by undoing the steps performed during create phase (step 2).
+   For clarity, the runtime MUST make a best-effort attempt to delete all resources that were create in step 2.
+   The phrase "best-effort" is to allow for situations where some other processes, or container, is "holding on" to a resource and the runtime is therefore unable to delete it.
```

Comment:
I dont really like "best-effort" approach here. I think we should just return an error for each resource that we can't delete and let the runtime caller decide.
Or perhaps we should reword this as `MUST delete all unreferenced resources that was created`.